### ISSUE TITLE
Skip privilege check in Information schema

### ISSUE THREAD

**Issue by yiwen92** [pingcap/tidb#29675]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
USE information_schema; 
CREATE user 'daily_dba'; 
GRANT Insert ON *.* TO 'daily_dba'@'%' WITH GRANT OPTION;
```
### 2. What did you expect to see? (Required)
exec good without error，just like mysql 8.0

### 3. What did you see instead (Required)
raise error: (8121, 'privilege check fail')

### 4. What is your TiDB version? (Required)

Release Version: v5.0.x and v4.0.x

**Comment by yiwen92** [comment]
this will affect version after 5.1.0, the root cause is no db info in processing process.

**Comment by tiancaiamao** [comment]
I checked master and master 6512fe5a497299f3b856988a744067496742f58c is not affected

**Comment by tiancaiamao** [comment]
This is fixed by https://github.com/pingcap/tidb/pull/23946, for the reason in https://github.com/pingcap/tidb/issues/24142

> It should be if dbName == "" && stmt.Level.Level != ast.GrantLevelGlobal {. There are also no tests on visitInfo for GrantLevelGlobal.

@morgo Would like to cherry-pick that fix to the old branch?

**Comment by morgo** [comment]
> This is fixed by #23946, for the reason in #24142
> 
> > It should be if dbName == "" && stmt.Level.Level != ast.GrantLevelGlobal {. There are also no tests on visitInfo for GrantLevelGlobal.
> 
> @morgo Would like to cherry-pick that fix to the old branch?

I chatted with @yiwen92 about it. It is not quite straight forward to cherry pick (it is based on other changes for dynamic privileges) but it should be easy enough to write a patch specific to 5.0.
